src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -6579,55 +6579,87 @@ def test_no_halo_for_discontinous(tmpdir, f90, f90flags):
         assert utils.code_compiles("dynamo0.3", psy, tmpdir, f90, f90flags)
 
 
-def test_halo_for_discontinuous(tmpdir, f90, f90flags):
-    ''' Test that we create halo exchange call when our loop iterates
-    over owned cells (e.g. it writes to a discontinuous field), we
-    read from a continous field, there are no stencil accesses, but
-    we do not know anything about the previous writer. As the previous
-    writer may have been over dofs we could have dirty annexed dofs
-    so need to add a halo exchange. '''
-    _, info = parse(os.path.join(BASE_PATH,
-                                 "1_single_invoke_w3.f90"),
-                    api="dynamo0.3")
-    psy = PSyFactory("dynamo0.3").create(info)
-    result = str(psy.gen)
-    print result
-    assert "IF (f1_proxy%is_dirty(depth=1)) THEN" in result
-    assert "CALL f1_proxy%halo_exchange(depth=1)" in result
-    assert "IF (f2_proxy%is_dirty(depth=1)) THEN" in result
-    assert "CALL f2_proxy%halo_exchange(depth=1)" in result
-    assert "IF (m1_proxy%is_dirty(depth=1)) THEN" in result
-    assert "CALL m1_proxy%halo_exchange(depth=1)" in result
+def test_halo_for_discontinuous(tmpdir, f90, f90flags, monkeypatch):
+    '''This test checks the case when our loop iterates over owned cells
+    (e.g. it writes to a discontinuous field), we read from a
+    continuous field, there are no stencil accesses, but we do not
+    know anything about the previous writer.
+
+    As we don't know anything about the previous writer we have to
+    assume that it may have been over dofs. If so, we could have dirty
+    annexed dofs so need to add a halo exchange (for the three
+    continuous fields being read (f1, f2 and m1). This is the case
+    when psyclone.config.COMPUTE_ANNEXED_DOFS is False.
+
+    If we always iterate over annexed dofs by default, our annexed
+    dofs will always be clean. Therefore we do not need to add a halo
+    exchange. This is the case when
+    psyclone.config.COMPUTE_ANNEXED_DOFS is True.
 
-    if utils.TEST_COMPILE:
-        # If compilation testing has been enabled
-        # (--compile --f90="<compiler_name>" flags to py.test)
-        assert utils.code_compiles("dynamo0.3", psy, tmpdir, f90, f90flags)
+    '''
+    for annexed in [False, True]:
```

Comment:
Again, another issue?

Reply:
I think we should do it for `annexed` in this issue otherwise it will probably never happen (and I *think* it will shorten our test code so it's worth doing).

Reply:
OK, done.